# Task 1.a: Write the hypotheses explicitly

1.Null ($H_{0}$): Popularity rating has no effect on the chance of winning an award.
<br>
Alternative ($H_{1}$): A higher popularity rating increases the chance of winning an award.

2.Null ($H_{0}$): Explicitness of a song has no effect on the chance of winning an award.
<br>
Alternative ($H_{1}$): Explicit songs are less likely to win an award.

3.Null ($H_{0}$): Tempo has no effect on the chance of winning an award.
<br>
Alternative ($H_{1}$): Songs with a higher tempo are more likely to win an award.

# Task 1.b: Choose and justify approporiate test

**1.Regression Coefficient Test**
<br>
**Regression Coefficient Test** are used to determine if a **predictor variable** has a **statistically significant relationship** with a **dependent variable**. Since **Popularity Rating** is a **Continuous Variable** it will be our **predictor variable** and **Grammy Winner**, a **Categorical Variable** will be our **dependent variable**. This will test whether popularity significantly predicts the probability of winning.

**2.Chi-Square Test**
<br>
**Chi-Square Test** are used on **categorical data** to determine if there is a **significant relationship** between **two variables**, or if your observed data **matches** an **expected distribution**. We will compare two categorical variables in **Explicitness** and **Grammy Winner** to see whether the two have a relationship.

**3.Regression Coefficient Test**
<br>
**Regression Coefficient Test** are used to determine if a **predictor variable** has a **statistically significant relationship** with a **dependent variable**. Since **Tempo** is a **Continuous Variable** it will be our **predictor variable** and **Grammy Winner**, a **Categorical Variable** will be our **dependent variable**. This will test whether tempo significantly predicts the probability of winning.

# Task 1.c: Test Statistic, P-value, and Comparison

In [5]:
import pandas as pd
import statsmodels.api as sm
from scipy.stats import chi2_contingency
df = pd.read_csv("newdata.csv")

# Regression Coefficient Test of Popularity & Winner:

In [28]:
x = sm.add_constant(df['popularity'])
y = df['winner']

model_popularity = sm.OLS(y, x).fit()

z_stat = model_popularity.tvalues['popularity']
p_value = model_popularity.pvalues['popularity']
alpha = 0.05

print("Regression Coefficient Test of Popularity & Winner: ")
print(f"Test Statistic (z): {z_stat:.3f}")
print(f"p-value: {p_value:.4f}")
print(f"Significance Level (α): {alpha}")

Regression Coefficient Test of Popularity & Winner: 
Test Statistic (z): 13.018
p-value: 0.0000
Significance Level (α): 0.05


**Since the p-value of 0.0000 is less than our significance level of 0.05 we reject the null hypothesis of "Popularity rating has no effect on the chance of winning an award."**
<br>
**Popularity significantly predicts the chance of winning an award. In our dataset, songs with higher popularity ratings are much more likely to win awards.**

# Chi-Square test of Explictness & Winner:

In [24]:
contingency = pd.crosstab(df['explicit'], df['winner'])
chi2, p_value, dof, expected = chi2_contingency(contingency)
alpha = 0.05

print("Chi-Square Test of Explictness & Winner:")
print(f"Chi-Square Statistic: {chi2:.3f}")
print(f"p-value: {p_value:.4f}")
print(f"Significance Level (α): {alpha}")

Chi-Square Test of Explictness & Winner:
Chi-Square Statistic: 1.865
p-value: 0.1720
Significance Level (α): 0.05


**Since the p-value of 0.1720 is greater than our significance level of 0.05 we fail to reject the null hypothesis of "Explicitness of a song has no effect on the chance of winning an award."**
<br>
**There is no significant relationship between explicitness and award outcome. In our dataset, explicit songs are neither more or less likely to win awards.**

# Regression Coefficient Test of Tempo & Winner:

In [26]:
x = sm.add_constant(df['tempo'])
y = df['winner']

model_popularity = sm.OLS(y, x).fit()

z_stat = model_popularity.tvalues['tempo']
p_value = model_popularity.pvalues['tempo']
alpha = 0.05

print("Regression Coefficient Test of Tempo & Winner: ")
print(f"Test Statistic (z): {z_stat:.3f}")
print(f"p-value: {p_value:.4f}")
print(f"Significance Level (α): {alpha}")

Regression Coefficient Test of Tempo & Winner: 
Test Statistic (z): -0.484
p-value: 0.6287
Significance Level (α): 0.05


**Since the p-value of 0.6287 is greater than our significance level of 0.05 we fail to reject the null hypothesis of "Tempo has no effect on the chance of winning an award."**
<br>
**Tempo does not significantly affect the chance of winning an award. Faster or slower songs have similar likelihoods of winning.**